### PR TITLE
Fix forking state from memory

### LINKED ISSUE #406: Incorrect exception in memory.py
https://github.com/trailofbits/manticore/blob/45d5b9f6d24ea69ac2238821a99c068452ab4295/manticore/core/memory.py#L957

`ForkState` is defined in state.py, which depends on memory.py

### PR DESCRIPTION
**Pull request by japesinator**
Fixes #406, but uses a deferred circular import to do so

### CODE REVIEW

File: manticore/core/state.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import logging
 from collections import OrderedDict
 
-from .smtlib import solver
+from .smtlib import solver, Bool
```
Comment (feliam) [inline manticore/core/state.py]: Bool is now used ad was not before?

File: manticore/core/memory.py [hunk 2]
```diff
@@ -954,7 +954,8 @@ def read(self, address, size):
                 condition = False
                 for base in e.solutions:
                     condition = Operators.OR(address == base, condition )
-                raise ForkState(condition)
+                from .state import ForkState
```
Comment (feliam) [inline manticore/core/memory.py]: move up ?

File: manticore/core/state.py (hunk 1 above)
Comment (japesinator) [inline manticore/core/state.py]: Bool was always used, it just stopped being imported for a bit (https://github.com/trailofbits/manticore/blob/42b462fc6c2c9ee66393991d692215475b09fcd4/manticore/core/state.py#L58)

File: manticore/core/memory.py (hunk 2 above)
Comment (japesinator) [inline manticore/core/memory.py]: If we import initially, we fail because of circular imports.

**Pull request merged**